src/character.py

```diff
@@ -1,15 +1,11 @@
-from controller import Controller
-from game import Game
 from typing import List
 
 class Character(object):
-    player_num = 0
-    def __init__(self, game: Game, cpu: bool, controller: Controller):
-        self.game = game
-        self.cpu = cpu
-        self.controller = controller
-        self.player_num = Character.player_num
-        Character.player_num += 1
+    def __init__(self, game: 'Game', controller: 'Controller'):
```

Comment:
what is the controller do/what is it? I see the import statement is gone, what's type 'Controller'?

Reply:
It's the "controller" from the MVC pattern, so it'll just handle input from the user. I figured it makes it a bit more flexible if 2 characters were to want to input things differently (e.g. player 1 wants to use a CLI while player 2 wants to use a GUI).

Reply:
As to explain why the import statement is gone, I was having circular import problems and Python doesn't seem to have include guards like C++. Since I was just using the import for type-hinting, I found that the best way to handle this is to not actually import but instead [just change the type to a string](https://stackoverflow.com/questions/39740632/python-type-hinting-without-cyclic-imports).